src/weapons/Weapon.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+    Created by Maitiu Morton 2/1/2017
+        Edited by DericM 3/8/2017
+*/
+#include "Weapon.h"
+#include <iostream>
+
+Weapon::Weapon(std::string type, int range, int damage, int clip, int clipMax, int ammo,int rAOE,
+        int reloadSpeed, int fireRate, bool isReadyToFire)
+        :type(type), range(range), damage(damage), ammo(ammo), rAOE(rAOE), reloadSpeed(reloadSpeed),
+        reloadTick(0), reloadDelay(200), fireRate(fireRate), fireTick(0), isReadyToFire(isReadyToFire){
+
+}
+
+Weapon::Weapon(const Weapon& w) : type(w.type), range(w.range), damage(w.damage), ammo(w.ammo),
+        rAOE(w.rAOE), reloadSpeed(w.reloadSpeed), reloadTick(w.reloadTick), reloadDelay(w.reloadDelay),
+        fireRate(w.fireRate), fireTick(w.fireTick), isReadyToFire(w.isReadyToFire) {
+
+}
+
+
+int Weapon::getClip(){
```

Comment:
getters should have their methods declared const since they don't change internal values